common/text/text_structure.cc

```diff
@@ -194,6 +194,11 @@ void TextStructureView::FocusOnSubtreeSpanningSubstring(int left_offset,
   VLOG(2) << __FUNCTION__ << " at " << left_offset << " +" << length;
   const int right_offset = left_offset + length;
   TrimSyntaxTree(left_offset, right_offset);
+  // Don't let the syntax tree be empty.
+  // Always return a tree with one node.
+  if (syntax_tree_ == nullptr) {
+    syntax_tree_ = MakeNode();
```

Comment:
This _looks_ like it it is a quick fix that works around a problem that probably  could've been caught earlier if some test for `syntax_tree_ == nullptr` had happend more upstream.
Did you explore this possibility ? If it was not feasible, why ?

This does look like a very specific situation that is fixed by this, but looking more upstream could probably eliminate a whole class of potential issues.

Reply:
Yes, you're right.
In this case (that is, the single `verilog_syntax` directive), the problem starts with the `TextStructureView::TrimSyntaxTree` function, which removes the prolog and epilog from the `syntax_tree`. In that case, there is no node or leaf associated with the original code in the tree, so the `TextStructureView::TrimSyntaxTree` function that uses `verible::TrimSyntaxTree` returns `nullptr` (because it removes the entire epilog and prolog). I can change the `verible::TrimSyntaxTree` function so that it is the one that never returns the tree as nullptr, only if `ZoomSyntaxTree` returns nullptr then this one will return a tree with one node (kUntagged).

What do you think about it @hzeller ?

Reply:
mmh, changing `TrimSyntaxTree()`, maybe even returning a bool would probably not improve the situation much.

So maybe for now, your solution is ok. Maybe add a comment pointing out that this can happen when TrimSyntaxTree() yields a nullptr syntax tree.

Reply:
Added comment